### PR TITLE
fix(INT-339): marketo response handler to handle static list removed and skipped statuses

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> [PR](https://github.com/rudderlabs/rudder-transformer/pull/2606) continued

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/marketo/util.js [hunk 1]
```diff
@@ -122,17 +123,27 @@ const marketoApplicationErrorHandler = (marketoResponse, sourceMessage, destinat
...
+          (errorCode) => !ERROR_CODE_TO_PASS.includes(errorCode),
+        );
+        if (!filteredErrorCode) {
+          return;
+        }
+        if (MARKETO_THROTTLED_CODES.includes(filteredErrorCode)) {
+          statusCode = 429;
+        } else if (MARKETO_RETRYABLE_CODES.includes(filteredErrorCode)) {
+          statusCode = 500;
+        }
```
Comment (shrouti1507) [inline src/v0/destinations/marketo/util.js:146]: throw new InstrumentationError(
        `Request failed during: ${sourceMessage}, error: ${JSON.stringify(reasons)}`, <-- lets add some default error message here as well

**Review by shrouti1507 — APPROVED** [review]
LGTM

File: src/v0/destinations/marketo/util.js (hunk 1 above)
Comment (shrouti1507) [reply src/v0/destinations/marketo/util.js:146]: Also, why 601 is retryable here. In marketo bulk upload we marked it abortable

Comment (anantjain45823) [reply src/v0/destinations/marketo/util.js:146]: What do we mean by default error message ?
Source Message is already there and due its complexity we can't say anything by default

Comment (anantjain45823) [reply src/v0/destinations/marketo/util.js:146]: Added a default message of Reasons not found

Comment (shrouti1507) [reply src/v0/destinations/marketo/util.js:146]: Also what is your take on the invalid access token status code?

Comment (anantjain45823) [reply src/v0/destinations/marketo/util.js:146]: It should result it in retryable error as it will generate a new access token and perform the operation

**Review by shrouti1507 — APPROVED** [review]

File: src/v0/destinations/marketo/util.js (hunk 1 above)
Comment (shrouti1507) [reply src/v0/destinations/marketo/util.js:146]: as per the offline chat, @anantjain45823 confirmed that invalid credentials does not return 601 status code.

**Review by yashasvibajpai — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**